zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java

```diff
@@ -152,24 +127,24 @@ void shouldPutComponentTemplate() {
   @Test
   void shouldAuthenticateWithBasicAuth() {
     // given
-    testClient.putUser("user", "^AHq>z@)&l;RJU=\"", List.of("admin"));
-    config.getAuthentication().setUsername("user");
-    config.getAuthentication().setPassword("^AHq>z@)&l;RJU=\"");
+    searchDB.testClient().putUser("user", "^AHq>z@)&l;RJU=\"", List.of("admin"));
+    searchDB.config().getAuthentication().setUsername("user");
+    searchDB.config().getAuthentication().setPassword("^AHq>z@)&l;RJU=\"");
 
     // when
     // force recreating the client
     final var authenticatedClient =
         new OpensearchClient(
-            config,
-            bulkRequest,
-            RestClientFactory.of(config, true),
-            indexRouter,
-            templateReader,
+            searchDB.config(),
+            searchDB.bulkRequest(),
+            RestClientFactory.of(searchDB.config(), true),
+            searchDB.indexRouter(),
+            searchDB.templateReader(),
             new OpensearchMetrics(new SimpleMeterRegistry()));
     authenticatedClient.putComponentTemplate();
```

Comment:
🔧 Maybe we want in the API actually a method to recreate the client with certain configs? Then we can get rid of the other properties exposed in the interface? 🤔

Reply:
If I understood your question correctly, we cannot do this. All those API methods `recordFactory()`, `indexRouter()`, etc methods are used in tests.

Reply:
PR moved here: https://github.com/camunda/camunda/pull/24986